src/codex_plus/main_sync_cffi.py

```
@@ -154,6 +246,399 @@ async def health():
+        raise HTTPException(status_code=500, detail=f"Failed to create agent: {str(e)}")
+
+@app.put("/agents/{agent_id}", response_model=AgentConfigResponse)
+async def update_agent(agent_id: str, agent_data: AgentConfigUpdate):
+    """Update existing agent configuration"""
+    logger.info(f"UPDATE AGENT REQUEST: {agent_id}")
+    try:
+        # Get existing configuration
+        existing_config = agent_middleware.config_loader.get_agent(agent_id)
+        if not existing_config:
+            logger.warning(f"Agent not found for update: {agent_id}")
+            raise HTTPException(status_code=404, detail=f"Agent '{agent_id}' not found")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)